### PR TITLE
Character class with 2 children classes

### PR DESCRIPTION
**Pull request by infanics2**
# Character abstract class which allows its children to fight

- Add Character abstract class which can be used as a parent class for all characters
- Add 2 children classes (Warrior and Archer)
- The instances of the children classes can fight each other and reduce their opponents' health based on the their power and skill value

### CODE REVIEW

File: src/main/java/abstraction/Archer.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package abstraction;
+
+public class Archer extends Character {
+    {
```
Comment (zaqqaz226) [inline src/main/java/abstraction/Archer.java:4]: Проинициализируй поля в конструкторе. Просто передавай значения, которые тебе нужны в super.

Comment (infanics2) [reply src/main/java/abstraction/Archer.java:4]: I hope I got your idea) Can you please check that again? Thank you!

Comment (zaqqaz226) [reply src/main/java/abstraction/Archer.java:4]: Nope) When you write it here, it looks like a field of a class. In line 14, you already have the constructor you need. Note that you are already using the constructor to set the name. Use this approach to set the other fields as well.

File: src/main/java/abstraction/Warrior.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        this.skill = 5;
+        this.intellect = 3;
+    }
+
+    public Warrior(String name) {
+        super(name);
+    }
+
+    public Warrior(String name, int power, int skill, int intellect) {
+        super(name);
```
Comment (zaqqaz226) [inline src/main/java/abstraction/Warrior.java:15]: User super constructor to set values.

File: src/main/java/abstraction/Archer.java (hunk 1 above)
Comment (infanics2) [reply src/main/java/abstraction/Archer.java:4]: Now, I got the idea, thanks)

File: src/main/java/abstraction/Warrior.java (hunk 2 above)
Comment (infanics2) [reply src/main/java/abstraction/Warrior.java:15]: Done

**Review by zaqqaz226 — APPROVED** [review]

**Pull request merged**